src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -967,22 +970,36 @@ bool AccountStoreSC<MAP>::ParseContractCheckerOutput(
         return false;
       }
 
-      Json::Value map_depth_json;
+      bool hasMap = false;
+
       if (root["contract_info"].isMember("fields")) {
         for (const auto& field : root["contract_info"]["fields"]) {
           if (field.isMember("vname") && field.isMember("depth") &&
-              field["depth"].isNumeric()) {
-            map_depth_json[field["vname"].asString()] = field["depth"].asInt();
+              field["depth"].isNumeric() && field.isMember("type")) {
+            metadata.emplace(
+                Contract::ContractStorage2::GetContractStorage()
+                    .GenerateStorageKey(addr, MAP_DEPTH_INDICATOR,
+                                        {field["vname"].asString()}),
+                DataConversion::StringToCharArray(field["depth"].asString()));
+            if (!hasMap && field["depth"].asInt() > 0) {
+              hasMap = true;
+            }
+            metadata.emplace(
+                Contract::ContractStorage2::GetContractStorage()
+                    .GenerateStorageKey(addr, TYPE_INDICATOR,
+                                        {field["vname"].asString()}),
+                DataConversion::StringToCharArray(field["type"].asString()));
+          } else {
+            LOG_GENERAL(WARNING, "Unexpected field detected" << checkerPrint);
```

Comment:
Might be useful to also log the field that caused the error, and not just the entire checkerPrint